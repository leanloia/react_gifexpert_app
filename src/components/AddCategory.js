import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit} action="submit">
        <input onChange={handleChange} type="text" value={inputValue} />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
