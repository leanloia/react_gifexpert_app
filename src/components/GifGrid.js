import React /*{ useState, useEffect }*/ from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {data.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
