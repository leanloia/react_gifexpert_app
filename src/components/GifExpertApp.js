import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Evangelion"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};

export default GifExpertApp;
