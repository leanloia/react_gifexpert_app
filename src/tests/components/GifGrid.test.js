import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en componente GifGrid", () => {
  const category = "boca sorete";

  test("debe mostrarse correctamente el componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    let wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes con useFetchGifs", () => {
    const gifs = [
      {
        id: "abd",
        url: "https://cualquiercosa.com.ar",
        title: "boca mierda",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    let wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    // corrobora que no existe el 'p' que tiene el 'loading'
    expect(wrapper.find("p").exists()).toBe(false);
    // corrobora del snapshot el item generado, y que tenga la misma cantidad que 'gifs'
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
