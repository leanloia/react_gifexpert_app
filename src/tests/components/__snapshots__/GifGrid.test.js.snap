// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGrid debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    boca sorete
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abd"
      key="abd"
      title="boca mierda"
      url="https://cualquiercosa.com.ar"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente GifGrid debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    boca sorete
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
